packages/cadl-ranch-specs/http/azure/core/basic/main.tsp

```diff
@@ -206,6 +206,120 @@ Expected response body:
 """)
 op listWithPage is global.Azure.Core.Foundations.Operation<{}, global.Azure.Core.Page<User>>;
 
+model ListItemInput<Type> {
+  @doc("The body of the input.")
+  @body
+  bodyInput: ListItemInputBody;
+
+  @doc("Another query parameter.")
+  @query
+  another?: Type;
+}
+
+#suppress "@azure-tools/typespec-azure-core/use-extensible-enum" "For testing"
+@doc("A fixed enum input paramter.")
+@global.Azure.Core.fixed
+enum ListItemInputFixedEnum {
+  @doc("The first enum value.")
+  First,
+
+  @doc("The second enum value.")
+  Second,
+}
+
+@doc("An extensible enum input paramter.")
+enum ListItemInputExtensibleEnum {
+  @doc("The first enum value.")
+  First,
+
+  @doc("The second enum value.")
+  Second,
+}
+
+@doc("The body of the input.")
+model ListItemInputBody {
+  @doc("The name of the input.")
+  inputName: string;
+}
+
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "For testing global.Azure.Core.Page"
+@scenario
+@doc("List with int32 parameter Azure.Core.Page<>.")
+@route("/int-page")
+@scenarioDoc("""
+Expected query parameter: api-version=2022-12-01-preview&another=1
+
+Expected body parameter: {"inputName": "Madge"}
+
+Expected response body:
+```json
+{
+  "value":[
+     {
+        "id": 1,
+        "name": "Madge",
+        "etag": "11bdc430-65e8-45ad-81d9-8ffa60d55b59"
+     }
+  ]
+}
+""")
+@get
+op listWithIntParameters is global.Azure.Core.Foundations.Operation<ListItemInput<int32>, global.Azure.Core.Page<User>>;
+
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "For testing global.Azure.Core.Page"
+@scenario
+@doc("List with fixed enum parameter Azure.Core.Page<>.")
+@route("/fixed-enum-page")
+@scenarioDoc("""
+Expected query parameter: api-version=2022-12-01-preview&another=Second
+
+Expected body parameter: {"inputName": "Madge"}
+
+Expected response body:
+```json
+{
+  "value":[
+     {
+        "id": 1,
+        "name": "Madge",
+        "etag": "11bdc430-65e8-45ad-81d9-8ffa60d55b59"
+     }
+  ]
+}
+""")
+@get
+op listWithFixedEnumParameters is global.Azure.Core.Foundations.Operation<
+  ListItemInput<ListItemInputFixedEnum>,
+  global.Azure.Core.Page<User>
+>;
+
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "For testing global.Azure.Core.Page"
+@scenario
+@doc("List with extensible enum parameter Azure.Core.Page<>.")
+@route("/extensible-enum-page")
+@scenarioDoc("""
+Expected query parameter: api-version=2022-12-01-preview&another=Second
+
+Expected body parameter: {"inputName": "Madge"}
+
+Expected response body:
+```json
+{
+  "value":[
+     {
+        "id": 1,
+        "name": "Madge",
+        "etag": "11bdc430-65e8-45ad-81d9-8ffa60d55b59"
+     }
+  ]
+}
+""")
+@get
+op listWithExtensibleEnumParameters is global.Azure.Core.Foundations.Operation<
+  ListItemInput<ListItemInputExtensibleEnum>,
+  global.Azure.Core.Page<User>
+>;
+
```

Comment:
Does the generation logic for .NET is different for an operation with query param with an pageable operation with query param? It looks like testing for different types of query param.

Reply:
Yes, we handle paging operation and normal operation separately.

Yeah it is not a must for us to test different types. I added just for more coverage.